files/templates/scrapeconfigs/additional-scrape-configs.template.yaml

```diff
@@ -527,36 +548,30 @@
   - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring, __meta_kubernetes_service_labelpresent_giantswarm_io_monitoring]
     regex: .*(true).*
     action: keep
-  - source_labels: [__meta_kubernetes_service_labelpresent_operated_prometheus]
-    regex: (true)
-    action: drop
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_port]
-    regex: (true)
-    action: keep
-  - source_labels: [__meta_kubernetes_service_annotationpresent_giantswarm_io_monitoring_path]
-    regex: (true)
-    action: keep
-  - regex: (.*)
-    target_label: __address__
-    replacement: [[ .APIServerURL ]]:443
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_path is present, we use it as the metrics path
+  - source_labels: [__meta_kubernetes_service_annotation_giantswarm_io_monitoring_path]
+    action: replace
+    target_label: __metrics_path__
+    regex: (.+)
+    # if __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port, we use it as the metrics port
+  - source_labels: [__address__, __meta_kubernetes_service_annotation_giantswarm_io_monitoring_port]
```

Comment:
How does this behave when `__meta_kubernetes_service_annotation_giantswarm_io_monitoring_port` is empty?

Reply:
Same as above: If the regex does not match, there will be no replacement: prometheus/prometheus@916dbd4/pkg/relabel/relabel.go#L211